aten/src/ATen/core/jit_type.h

```diff
@@ -622,8 +641,28 @@ struct CAFFE2_API TupleType : public Type {
     }
     return true;
   }
+
+  bool compare_names(const Type& rhs) const {
+    if(rhs.kind() != kind())
+      return false;
+    if (!hasNames() && !rhs.cast<TupleType>()->hasNames())
+      return true;
+    if (!hasNames() || !rhs.cast<TupleType>()->hasNames())
+      return false;
+    const auto & l_names = names();
```

Comment:
It is not necessary to reimplement vector equality. `return names() == rhs->names();`